packages/@react-aria/focus/src/focusSafely.ts

```diff
@@ -24,12 +24,12 @@ export function focusSafely(element: FocusableElement) {
   // the page before shifting focus. This avoids issues with VoiceOver on iOS
   // causing the page to scroll when moving focus if the element is transitioning
   // from off the screen.
-  const ownerDocument = getOwnerDocument(element);
+  const activeElement = getDeepActiveElement();
```

Comment:
In order to continue to work in same-origin iframe scenario I suggest to get `document`  instance in ` getDeepActiveElement` from `element.ownerDocument` (not to use global `document` instance).

Reply:
resolved